### PR TITLE
disttask: update task in framework

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #46258 

Problem Summary:
Framework should updateTask.

### What is changed and how it works?
 updateTask in framework only.
See comments newly added in task_table.go
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -640,32 +640,39 @@ func (stm *TaskManager) UpdateGlobalTaskAndAddSubTasks(gTask *proto.Task, subtas
 			return err
 		}
 		if se.GetSessionVars().StmtCtx.AffectedRows() == 0 {
-			retryable = false
-			return errors.New("invalid task state transform, state already changed")
+			rs, err := ExecSQL(stm.ctx, se, "select id from  mysql.tidb_global_task where id = %? and state = %?", gTask.ID, prevState)
```
Comment (ywqzzy) [inline disttask/framework/storage/task_table.go:643]: Maybe gtask has no update(currently add index/import into will always change gtask), we need to check if the task have changed its state by admin cancel.

Comment (D3Hunter) [reply disttask/framework/storage/task_table.go:643]: why query it again when `StmtCtx.AffectedRows()` is 0?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: disttask/framework/storage/task_table.go (hunk 1 above)
Comment (ywqzzy) [reply disttask/framework/storage/task_table.go:643]: > why query it again when `StmtCtx.AffectedRows()` is 0?

If some onNextStage didn't update gtask, the affectedrows() return 0. But if the state didn't change by other admin command, it's safe to insert subtasks.
This case will not occurred for current add index and import into.

**Review by D3Hunter — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**